### ISSUE TITLE
Conditional metaclass causes `type` to not be propagated correctly

### ISSUE THREAD

**Issue by anuraaga** [astral-sh/ty#4370]
### Summary

When a metaclass is defined conditionally, ty seems to lose `builtins.type` from the hierarchy, causing a match on `type()` to not match.

Minimal repro:
```python
class Meta(type): ...
MessageMeta = Meta if os.environ.get("X") else type

class Msg(metaclass=MessageMeta): ...

def f(arg: type[Msg]) -> None:
    match arg:
        case type(): pass
        case _: assert_never(arg)  # error: type[Msg] & ~type not Never
```

Integration case: 
https://github.com/bufbuild/protobuf-py/blob/main/src/protobuf/_registry.py#L84
https://github.com/bufbuild/protobuf-py/blob/main/src/protobuf/_message.py#L85

Passes on 0.0.73, fails on 0.0.74

### Version

ty 0.0.74 (00199f0aa 2026-08-22)

**Comment by dhruvmanila** [comment]
Thanks for the report! This seems to be originating from https://github.com/astral-sh/ruff/pull/27897

**Comment by carljm** [comment]
We don't support union metaclasses in general, and neither does any other type checker; I don't think that should be a priority to support.

That said, this union actually eagerly simplifies in ty to just `type[type]`, which we should be able to support as a metaclass.
